File: src/main/java/project/rpg/player/status/thing/Strength.java
package project.rpg.player.status.thing;

import org.bukkit.attribute.Attribute;
import org.bukkit.entity.Player;
import project.rpg.player.status.base.StatusBase;
import project.rpg.player.status.base.StatusName;

public class Strength extends StatusBase {   //힘

    public Strength() {
        super(StatusName.STRENGTH.getName());
    }

    @Override
    public void effect(Player player) {
        player.getAttribute(Attribute.GENERIC_ATTACK_DAMAGE).setBaseValue(2d+(this._value-10d)/2d);
    }

}

File: src/main/java/project/rpg/player/status/objects/Strength.kt
package project.rpg.player.status.objects

import kotlinx.serialization.Serializable
import org.bukkit.attribute.Attribute
import org.bukkit.entity.Player
import project.rpg.player.status.base.StatusBase
import project.rpg.player.status.base.StatusName

@Serializable
class Strength: StatusBase(StatusName.STRENGTH) {
    override fun effect(player: Player?) {
        player!!.getAttribute(Attribute.GENERIC_ATTACK_DAMAGE)!!.baseValue = 2.0 + (value - 10.0) / 2.0
    }
}